lib/sites-base.js

```diff
@@ -224,8 +224,25 @@ module.exports = {
         // Most common mistakes: protocol in the shortname, domain name in the shortname
         if (shortName.match(/:|\.|\s/)) {
           self.apos.notify(req, 'The short name of the site must not contain dots, a protocol or spaces. It is a short name like "nifty" (without quotes) <F2>and will be used as a "working name" for a temporary subdomain for your site until it is launched.', { type: 'error' });
-          return callback('invalid');
+          return callback(new Error('invalid'));
         }
+
+        const existing = await self
+          .find(req, {
```

Comment:
findForEditing

Reply:
Ensures you don't miss anything that would be hidden from the dashboard homepage for whatever reason but available in the manager/editor

Reply:
with that we can safely remove

```js
.trash(false)
.published(true)
```

?

Reply:
Actually I think we don't want those behaviors anyway for this particular check, so remove them, yeah, but only because we *do* want to prevent shortname conflicts with unpublished sites.

As for sites in the trash, you can call `self.addTrashSuffixFields([ 'shortName' ])` in `afterConstruct` of sites to automatically deduplicate that field when it is in the trash so there is no conflict unless you try to take it out.

Reply:
`addTrashSuffixFields` works well for newly trashed sites:

```
shortName: 'test4-deduplicate-clptyjvri007rwo1uhwn9vpka',
```

but not for existing sites in the trash (ie. sites that were put in the trash before `addTrashSuffixFields` were added.

So there could be a shortName conflict with those, but I assume that's not an issue.